addon/session-stores/adaptive.js

```diff
@@ -137,5 +140,10 @@ export default Base.extend({
   */
   clear() {
     return this.get('_store').clear();
-  }
+  },
+
+  _cookieDataChanged: observer('cookieDomain', 'cookieName', 'cookieExpirationTime', function() {
+    const data = this.getProperties('cookieDomain', 'cookieName', 'cookieExpirationTime');
+    return this.get('_store').persist(data);
+  })
```

Comment:
Could we do that with getters and setters instead of an observer? E.g.

```js
cookieDomain:  computed({
  get() {
    return this._cookieDomain;
  },
  set(_, value) {
    this._cookieDomain = value;
    this.rewriteCookie();
  }
}),
```

Also it should be sufficient to only define that on the cookie store. We just need to make sure that the cookie settings are defined as aliases on the `_store` then.